replay_log.py

```diff
@@ -61,12 +72,25 @@ def populate_server_with_log(log_file):
 def replay_log(log_file, request_to_file_uuid):
   fd = open(log_file, 'r')
   for line in fd:
-    request_source_info, request_content, reply = line.split('"')
-    request_source_ip = request_source_info.split()[0]
+    request_source, request_content, request_size, concurrent = line.split('\t')
+    print 'request_source: ' + str(request_source)
+    print 'request_content: ' + str(request_content)
+    print 'request_size: ' + str(request_size)
+    print 'concurrent: ' + str(concurrent)
     file_uuid = request_to_file_uuid[request_content]
-    succeed = read_file(file_uuid, request_source_ip)
-    if not succeed:
-      raise ValueError('request failed with file uuid: ', file_uuid)
+
+    # If concurrent, run concurrently with delay
+    if concurrent.strip() == 'C':
+      delay = float(request_size) / 1000
+      # cap delay at 3 seconds
+      if delay > 3:
+        delay = 3
+      # succeed = read_file(file_uuid, request_source, delay)
+      Process(target=read_file, args=(file_uuid, request_source, delay)).start()
```

Comment:
Will it wait until all concurrent requests are being made to process the log entry?